packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -252,15 +266,16 @@ export abstract class TemplateLinter<
     return { doc: undefined, uri: undefined, json: undefined };
   }
 
-  /** Load the variableDefinition file and return a map of variableName -> variableType + isArray, or undefined if
+  /** Load the variableDefinition file and return a map of variableName -> variableType + isArray + node, or undefined if
    * variablesDefinition isn't specified or can't be read.
    */
-  protected async loadVariableTypesForTemplate(
-    templateInfo: JsonNode
-  ): Promise<Record<string, { type: string; isArray: boolean }> | undefined> {
-    const { json: varJson } = await this.loadTemplateRelPathJson(templateInfo, ['variableDefinition']);
-    if (varJson && varJson.type === 'object' && varJson.children && varJson.children.length > 0) {
-      const variableTypes: Record<string, { type: string; isArray: boolean }> = {};
+  protected async loadVariableTypesForTemplate(templateInfo: JsonNode): Promise<{
+    doc: Document | undefined;
```

Comment:
Needed to refactor to also return the variableDefinition doc, for later doing `relatedInformation` on diagnostics.